examples/rust/battleship/src/main.rs

```diff
@@ -12,6 +12,348 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use battleship_core::*;
+use lib::*;
+use std::{borrow::Borrow, io::*};
```

Comment:
Looks like you can drop `Borrow`?
Would prefer that we avoid using `::*`, since it makes it less obvious what was pulled in.